mlir/tools/mlir-miopen-lib/mlir-miopen-lib.h

```diff
@@ -0,0 +1,85 @@
+//===-------------- mlir-miopen-lib.h - C API ---------------------*- C -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM
+// Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_C_MIIR_LIBMIIR_H
+#define MLIR_C_MIIR_LIBMIIR_H
+
+#include <stddef.h>
+
+#define MIIR_VERSION_FLAT 0
+
+enum MiirStatus {
+  MIIR_SUCCESS = 0,
+  MIIR_INVALID_PARAM,
+  MIIR_INVALID_MODULE,
+  MIIR_BUILD_FAILURE
+};
+typedef enum MiirStatus MiirStatus;
+
+/*! @brief The MLIR handle used for lowering and code generation
+ */
+typedef void *MiirHandle;
+
+/*! @brief Create the MLIR handle according to options string
+ *  @param options Command-line options as a string
+ *  @return        MLIR handle
+ */
+extern "C" MiirHandle miirCreateHandle(const char *options);
+
+/*! @brief Lower the MLIR module to c++ code
+ *  @param handle   MLIR handle
+ */
+extern "C" MiirStatus miirLowerCpp(MiirHandle handle);
+
+/*! @brief Populate Conv2d implicitgemm host code for MIOpen
+ *  @param handle   MLIR handle
+ *  @return         Source string
+ */
+extern "C" const char *miirGenIgemmSource(MiirHandle handle);
+
+/*! @brief Populate Conv2d implicitgemm header code for MIOpen
+ *  @param handle   MLIR handle
+ *  @return         Header string
+ */
+extern "C" const char *miirGenIgemmHeader(MiirHandle handle);
+
+/*! @brief Populate Conv2d implicitgemm compilation flags for MIOpen
+ *  @param handle   MLIR handle
+ *  @return         Compilation flags string
+ */
+extern "C" const char *miirGenIgemmCflags(MiirHandle handle);
+
+/*! @brief Lower the MLIR module to binary code
+ *  @param handle MLIR handle
+ */
+extern "C" MiirStatus miirLowerBin(MiirHandle handle);
+
+/*! @brief Populate Conv2d implicitgemm hsaco code object
+ *  @param handle MLIR handle
+ *  @param buffer Binary buffer holds hsaco code
+ *  @param size Size of the binary buffer
+ */
+extern "C" MiirStatus miirGenIgemmBin(MiirHandle handle, char **buffer,
+                                      size_t *size);
```

Comment:
`...*buffer... ` If it is nullptr, function updates `*size` with the size of the binary. Otherwise it copies the binary to the `*buffer` (`size` is irrelevant in this case), right?

Reply:
I think so. WIP on this. Will ping you when I fix this.

Reply:
Done in 2b0887c. Thanks